pkg/apis/cr/v1alpha1/types.go

```diff
@@ -307,3 +308,93 @@ type ProfileList struct {
 	metav1.ListMeta `json:"metadata"`
 	Items           []*Profile `json:"items"`
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+
+// RepositoryServer manages the lifecycle of Kopia Repository Server within a Pod
+type RepositoryServer struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              RepositoryServerSpec   `json:"spec"`
+	Status            RepositoryServerStatus `json:"status"`
+}
+
+// RepositoryServerSpec is the specification for the RepositoryServer
+type RepositoryServerSpec struct {
+	Storage    Storage    `json:"storage,omitempty"`
+	Repository Repository `json:"repository,omitempty"`
+	Server     Server     `json:"server,omitempty"`
```

Comment:
Should we set `omitempty` for all of there? I think `Repository` and `Storage` are required. Should we mark these as required as well in CRD definition?